client/src/components/DetailView.js

```diff
@@ -150,19 +143,47 @@ class DetailViewWithoutI18n extends Component {
                           <b>
                             <Trans>Last registered:</Trans>
                           </b>{" "}
-                          {this.formatDate(this.props.addr.lastregistrationdate)}
+                          {Helpers.formatDate(
+                            this.props.addr.lastregistrationdate,
+                            {
+                              year: "numeric",
+                              month: "short",
+                              day: "numeric",
+                            },
```

Comment:
This object of date options seems to be used a bunch within the DetailView component. One suggestion for making this more DRY ("Don't Repeat Yourself") would be to actually save this object as a variable within the file, much like you do with `locale`. That way, you can just call it each time that you need it without spelling everything out again.